### DISCUSSION TITLE
What is lacking in order to migrate to gitoxide

### DISCUSSION THREAD

**Discussion by nyurik** [extrawurst/gitui#1709] [Ideas]
[gitoxide](https://github.com/Byron/gitoxide) is a (difficult!) attempt to rewrite all of git's functionality in pure Rust. As always, in order for anything to develop into a better state, it needs to be needed/used by others.  Is there a path/plan to support gitoxide in the future, probably as an opt-in feature at first?  What functionality is missing on the gitoxide side in order for this to happen?

**Comment by extrawurst** [comment]
I am not following gitoxide close enough to know what's missing. Blanketly said most of libgit2 functionality is used by gitui. so that would be a way to check whether all these APIs are provided by gitoxide. 

If this is ever the case I am happy to switch away from interfacing to libgit2 for sure 💪

**Comment by Byron** [comment]
> What functionality is missing on the gitoxide side in order for this to happen?

I have done an initial analysis to get a rough understanding here: https://github.com/Byron/gitoxide/issues/1078 .

In short, a lot of the higher-level features are missing, but I also think it's no problem to provide them once they are on the roadmap.
However, it's clear that this is a far-out proposition, and I have a feeling that once `gitui` can use `gitoxide`, it truly has a become a `git2` replacement. Give it a couple of years 😅.
